packages/babel-helpers/src/helpers/regenerator.ts

```diff
@@ -166,389 +171,223 @@ export default function /* @no-mangle */ _regenerator() {
   function makeInvokeMethod(
     innerFn: Function,
     self: unknown,
-    context: Context,
+    tryLocsList: TryLocs[],
   ) {
     var state = GenState.SuspendedStart;
 
-    return function invoke(method: "next" | "throw" | "return", arg: any) {
+    function invoke(
+      _methodName: "next" | "throw" | "return",
+      _method: OperatorType.Next | OperatorType.Throw | OperatorType.Return,
+      _arg: any,
+    ) {
       if (state === GenState.Executing) {
-        throw new Error("Generator is already running");
-      }
-
-      if (state === GenState.Completed) {
-        if (method === "throw") {
-          throw arg;
+        throw Error("Generator is already running");
+      } else if (done) {
+        if (_method === OperatorType.Throw) {
+          Context_dispatchExceptionOrFinishOrAbrupt(_method, _arg);
         }
-
-        // Be forgiving, per GeneratorResume behavior specified since ES2015:
-        // ES2015 spec, step 3: https://262.ecma-international.org/6.0/#sec-generatorresume
-        // Latest spec, step 2: https://tc39.es/ecma262/#sec-generatorresume
-        return { value: undefined, done: true };
       }
 
-      context.method = method;
-      context.arg = arg;
-
-      while (true) {
-        var delegate = context.delegate;
-        if (delegate) {
-          var delegateResult = maybeInvokeDelegate(delegate, context);
-          if (delegateResult) {
-            if (delegateResult === ContinueSentinel) continue;
-            return delegateResult;
+      method = _method;
+      arg = _arg;
+
+      while (!done || (_ = undefined)) {
```

Comment:
Does moving `_ = undefined` after the `while` make the code bigger? It seems like an optimization Terser should be able to make by itself.

Reply:
If we move it to after the `while`, it needs to be `if (done)  _ = undefined` because we have a break inside the loop.

https://github.com/babel/babel/pull/17268/files#diff-d95a593570d888bc7ef9ff1ec0157fa86ec3808e4b64dfcd80aa78748e2d8b0bR277-R279